### PR TITLE
[SDFAB-1031] Add ListenN4 method to allow UP initiated associations

### PR DESCRIPTION
**Pull request by EmanueleGallone**
- make golint skip test files
- Add `ListenN4()` method to allow UP to start association
- Add unit test for `ListenN4()`

Should enable the implementation of integrations tests, as described here: https://github.com/omec-project/upf-epc/issues/453

### CODE REVIEW

File: pkg/pfcpsim/pfcpsim.go [hunk 1]
```diff
@@ -121,6 +121,44 @@ func (c *PFCPClient) receiveFromN4() {
...
+	if err != nil {
+		return err
+	}
+
+	msg, err := message.Parse(buf[:n])
+	if err != nil {
+		return err
+	}
+
+	if _, ok := msg.(*message.AssociationSetupRequest); !ok {
```
Comment (EmanueleGallone) [inline pkg/pfcpsim/pfcpsim.go:143]: we're not saving UP's node ID anywhere, for now. I assume that we should do that in future

File: pkg/pfcpsim/pfcpsim.go [hunk 2]
```diff
@@ -121,6 +121,52 @@ func (c *PFCPClient) receiveFromN4() {
...
+
+		return err
+	}
+
+	msg, err := message.Parse(buf[:n])
+	if err != nil {
+		return err
+	}
+
+	if _, ok := msg.(*message.AssociationSetupRequest); !ok {
```
Comment (EmanueleGallone) [inline pkg/pfcpsim/pfcpsim.go:151]: we're not saving UP's node ID anywhere, for now. I assume that we should do that in future

File: pkg/pfcpsim/pfcpsim.go [hunk 3]
```diff
@@ -121,6 +121,52 @@ func (c *PFCPClient) receiveFromN4() {
...
 }
 
+// ListenN4 allows for UP initiated PFCP associations. It waits for a connection and for an association request from UP.
+func (c *PFCPClient) ListenN4() error {
+	conn, err := net.ListenUDP("udp", &net.UDPAddr{IP: net.ParseIP("0.0.0.0"), Port: PFCPStandardPort})
+	if err != nil {
+		return err
+	}
+	// Expect a request in 10 seconds.
+	if err = conn.SetDeadline(time.Now().Add(10 * time.Second)); err != nil {
```
Comment (EmanueleGallone) [inline pkg/pfcpsim/pfcpsim.go:131]: I'm wondering if we could pass the connection deadline as a parameter

File: pkg/pfcpsim/pfcpsim.go [hunk 4]
```diff
@@ -92,6 +94,19 @@ func (c *PFCPClient) sendMsg(msg message.Message) error {
...
+		if _, err := c.conn.WriteToUDP(b, rAddr); err != nil {
+			return err
+		}
+
+		return nil
+	}
+
 	if _, err := c.conn.Write(b); err != nil {
 		return err
 	}
```
Comment (EmanueleGallone) [inline pkg/pfcpsim/pfcpsim.go:112]: What about using always `WriteToUDP` method, for consistency? When dialing up a UDP address we save the remote address.

**Comment by gab-arrobo** [comment]
@EmanueleGallone, is this PR still valid? If so, can you please rebase it? Thanks!

**Pull request closed without merging**